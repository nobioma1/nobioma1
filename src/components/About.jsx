import React from 'react';
import Layout from './Layout';

const About = () => {
  return <Layout bg="#FFFFFF" title="About" id="about"></Layout>;
};

export default About;
