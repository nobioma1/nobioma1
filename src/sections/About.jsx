import React from 'react';
import styled from 'styled-components';
import ScrollAnimation from 'react-animate-on-scroll';

import Layout from '../components/Layout';
import data from '../data';
import avatar from '../assets/noble.jpg';
import icons from '../utils/icons';

const AboutContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;

  h2 {
    margin-top: 15px;
    font-size: 3rem;
  }

  p {
    font-size: 2.2rem;
    text-align: justify;
    line-height: 1.8;
  }
`;

const AboutInfo = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 5px;

  & > div {
    width: 200px;
    height: 200px;
    margin: 10px 0;

    img {
      width: 100%;
      border-radius: 50%;
    }
  }

  @media (min-width: 800px) {
    flex-direction: row;
    margin-top: 0px;
    align-items: center;
    justify-content: space-between;

    & > div {
      width: 25%;
      margin: 10px 0;

      img {
        width: 100%;
        border-radius: 50%;
      }
    }

    & > p {
      width: 70%;
    }
  }
`;

const Lang = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;

  & > div {
    border-radius: 3px;
    margin: 5px 0;
    border: 1px solid black;
    width: 45%;
    font-size: 2rem;
    padding: 5px;
    font-weight: 500;
    display: flex;
    align-items: center;
    transition: 0.4s ease-in-out;

    & > div {
      width: 25px !important;
      height: 25px;
      margin-right: 10px;

      & > svg {
        width: 25px;
        height: 25px;
      }
    }

    &:hover {
      transform: scale(1.05);
    }
  }
`;

const About = () => {
  return (
    <Layout bg="#18171c" title="About" id="about" centerY>
      <AboutContainer>
        <ScrollAnimation animateIn="fadeIn" animateOnce>
          <AboutInfo>
            <div>
              <img src={avatar} alt="Noble's Avatar" />
            </div>
            <p>{data.bio}</p>
          </AboutInfo>
          <h2>Skills</h2>
          <Lang>
            {data.skills.map((skill, index) => (
              <div key={index}>
                {icons(skill)}
                <p>{skill}</p>
              </div>
            ))}
          </Lang>
        </ScrollAnimation>
      </AboutContainer>
    </Layout>
  );
};

export default About;
