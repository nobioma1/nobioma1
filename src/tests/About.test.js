import React from 'react';
import renderer from 'react-test-renderer';

import About from '../components/About';

describe('About', () => {
  it('About Should Match Snapshot', () => {
    const tree = renderer.create(<About />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
