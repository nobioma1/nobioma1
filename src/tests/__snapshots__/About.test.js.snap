// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About About Should Match Snapshot 1`] = `
<div
  className="sc-iwsKbI ihjwvd"
>
  <div
    className="sc-gZMcBi kKDAKj"
  >
    <div
      className="sc-bdVaJa cdJaUk"
    >
      <img
        alt="Coding"
        src="coding.svg"
      />
    </div>
  </div>
  <div
    className="sc-gqjmRU jOGXJO"
  >
    <h1
      className="sc-bwzfXH irohca"
    >
      About Me
    </h1>
    <p>
      Hello, I'm a Full-Stack Engineer based in Nigeria. 
    Loving every bit of what I do, creating solutions to real problems that
    can be solved using technology and also improving already existing solutions.
    </p>
    <p>
      I'm good at:
    </p>
    <div
      className="sc-VigVT cTLLZs"
    >
      <p>
        HTML
      </p>
      <p>
        CSS
      </p>
      <p>
        JavaScript
      </p>
      <p>
        React
      </p>
      <p>
        NodeJs
      </p>
    </div>
  </div>
</div>
`;
